File: cdmsbats/bookrecord.cpp
#include "bookrecord.h"

BookRecord::BookRecord()
{
    _digBaseAddress = NULL;
    _digChannel = NULL;
    _detCode = NULL;
}

void BookRecord::StoreValues(int32_t* Record)
{
    _digBaseAddress = Record[0];
    _digChannel = Record[1];
    _detCode = Record[2];
}

void BookRecord::PrintValues()
{
    cout << "Digitizer Base Address = 0x" << setbase(16) << _digBaseAddress << endl;
    cout << "Digitizer Channel = " << _digChannel << endl;
    cout << "Detector Code = " << _detCode << endl;
}

File: scripting/python/cdmsbats/BookRecord.py
# -*- coding: utf-8 -*-
"""
Created on Fri Apr 11 10:57:59 2014

@author: tdoughty1
"""

from DataRecord import DataRecord


class BookRecord(DataRecord):

    def __init__(self):

        self._digBaseAddress = None
        self._digChannel = None
        self._detCode = None

    def StoreValues(self, Record):

        self._digBaseAddress = Record[0]
        self._digChannel = Record[1]
        self._detCode = Record[2]

    def PrintValues(self):

        print "Digitizer Base Address = 0x%x" % self._digBaseAddress
        print "Digitizer Channel = %d" % self._digChannel
        print "Detector Code = %d" % self._detCode
